### ISSUE TITLE
Unknown error during query transaction. Please check JS console logs. 

### ISSUE THREAD

**Issue by digitalstudium** [grafana/loki#1230]
**Describe the bug**
I have logs in Loki with label `instance="example"` for last 3 hours. I want to grep errors in these logs. When I use such query in grafana with timerange 1, 2 or 3 hours: ``` {instance="example"} error``` everything is working as expected. But when I expand timerange to 24 hours in grafana, then it stucks and after timeout shows "Unknown error during query transaction. Please check JS console logs. " error. As I understands, it stucks because there is no logs with a timestamp earlier than 3 hours. 

**To Reproduce**
1. Loki: v0.4.0
2. Promtail: v0.4.0
3. Query: `{instance="example"} error` for timerange that goes beyond the timerange in which the logs exist.

**Expected behavior**
I see log records from Loki with "error" entry in grafana for the last **n** hours.

**Environment:**
 - Infrastructure:  bare-metal
 - Deployment tool: plain binary with config file

**Loki config and logs**
All timeouts in Loki config are big enough: 
```
server:
  http_listen_port: 3100
  http-read-timeout: 10m
  http-write-timeout: 10m

querier:
  query_timeout: 10m
```
After enabling debug level I see such messages in logs:
```
http: superfluous response.WriteHeader call from github.com/opentracing-contrib/go-stdlib/nethttp.(*statusCodeTracker).WriteHeader (status-code-tracker.go:19)

traceID=13bbce9d73075336 msg="GET /api/prom/query?direction=BACKWARD&limit=1000&regexp=(%3Fi)error ... (500) 35.035883882s Response: \"write tcp 10.33.29.110:3100->10.33.29.55:48534: i/o timeout\\n\" ws: false ...
```

**Comment by slim-bean** [comment]
Would you be able to make the same query using [logcli](https://github.com/grafana/loki/blob/master/docs/getting-started/logcli.md) to see if we can isolate Loki vs Grafana as the source of the issue?

Also, if you open your browsers developer tools and check the console does the error give any more detailed info? Thanks!

**Comment by digitalstudium** [comment]
> Grafana has default 30s timeout for datasource.
> 
> see https://grafana.com/docs/installation/configuration/#timeout

There is no issue with grafana timeout. When I set timeout to 5 minutes, then I get the same error after 5 minutes.

**Comment by digitalstudium** [comment]
> Would you be able to make the same query using [logcli](https://github.com/grafana/loki/blob/master/docs/getting-started/logcli.md) to see if we can isolate Loki vs Grafana as the source of the issue?
> 
> Also, if you open your browsers developer tools and check the console does the error give any more detailed info? Thanks!

In browser console, I see this error: 
Failed to load resource: the server responded with a status of 502 (Bad Gateway)
After increasing grafana timeout to 5 minutes, I see the same error after a bit more time.

Here is a full log error from loki:
Nov 05 05:34:22 lokivm-1 loki[125715]: level=warn ts=2019-11-05T13:34:22.828343861Z caller=logging.go:49 traceID=740d21f13c475319 msg="GET /api/prom/query?direction=BACKWARD&limit=1000&regexp=(%3Fi)error&query=%7Binstance%3D%22example%22%7D&start=1572874385123000000&end=1572960785123000000&refId=A (500) 38.297942619s Response: \"write tcp 10.33.29.110:3100->10.33.29.55:49392: i/o timeout\\n\" ws: false; Accept: application/json, text/plain, */*; Accept-Encoding: gzip, deflate; Accept-Language: en-US,en;q=0.9; User-Agent: Grafana/6.4.3; X-Forwarded-For: 10.33.29.55, 10.33.29.55; X-Grafana-Org-Id: 1; X-Rbt-Cli: Name=...; Ver=9.8.1;; "

**Comment by davkal** [comment]
Re the title of the issue: Grafana did a change in https://github.com/grafana/grafana/pull/19910 that surfaces the connection error. This will be released in Grafana 6.5 this month.
